sympy/functions/elementary/tests/test_piecewise.py

```diff
@@ -491,7 +501,36 @@ def test_as_expr_set_pairs():
         [((x - 2)**2, Interval(0, oo)), (0, Interval(-oo, 0, True, True))]
 
 
+@XFAIL
 def test_S_srepr_is_identity():
+    # this fails as follows:
```

Comment:
THIS is driving me crazy. Can anyone help me? When I have it print out what is going on in `_print_Piecewise` it looks like the first ExprCondPair is correct and that the seond one looks like an application of a function named ExprCondPair rather than the class.